### PR TITLE
Fix create_index/create_index_with_IOException issue caused by OpenSearch PR change

### PR DESCRIPTION
**Pull request by MitchellGale**
### Description
Fixes broken tests that were broke by [#8636](https://github.com/opensearch-project/OpenSearch/pull/8636).

Adds `setDefaultMediaType` to set default media type to be `XContentType.JSON` (previously was hard coded).
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java [hunk 1]
```diff
@@ -129,6 +130,7 @@ void is_index_exist_with_exception() throws IOException {
   @Test
   void create_index() throws IOException {
     String indexName = "test";
+    setDefaultMediaType(XContentType.JSON);
```
Comment (vamsi-amazon) [inline opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java:133]: Is this expected? How does this work when a new cluster is deployed.

File: opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java [hunk 2]
```diff
@@ -34,10 +35,7 @@
 import lombok.SneakyThrows;
 import org.apache.commons.lang3.reflect.FieldUtils;
 import org.apache.lucene.search.TotalHits;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.DisplayNameGeneration;
-import org.junit.jupiter.api.DisplayNameGenerator;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (matthewryanwells) [inline opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java:38]: Should this wildcard import be removed?

File: opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java (hunk 1 above)
Comment (Yury-Fridlyand) [reply opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java]: Before [8636](https://github.com/opensearch-project/OpenSearch/pull/8636) core used `JSON` hardcoded. Now it uses `DefaultMediaType` which is unset (`null`) by default.
You can re-run these tests on a branch to see the failure.

**Review by matthewryanwells — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

File: opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java (hunk 2 above)
Comment (MitchellGale) [reply opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java]: Thanks! Fixed.

File: opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java (hunk 1 above)
Comment (MitchellGale) [reply opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java]: I moved the setDefaultMediaType to a `BeforeAll`. but yes I believe it is the expected Media Type for these tests.

Comment (vamsi-amazon) [reply opensearch/src/test/java/org/opensearch/sql/opensearch/client/OpenSearchRestClientTest.java]: @Yury-Fridlyand  I looked into this yesterday, I couldn't figure out the need to introduce a new function for setting default type in core. Probably I should comment in 8636.

**Review by MaxKsyunz — APPROVED** [review]

**Comment by MaxKsyunz** [comment]
@MitchellGale does this need to be backported to 2.x?

**Comment by MitchellGale** [comment]
Yes please @MaxKsyunz

**Comment by MitchellGale** [comment]
Does adding the flag work now, or do I need to manually backport?

**Comment by Yury-Fridlyand** [comment]
I think auto-backport is good enough, but it is not yet backported on upstream

**Pull request merged**